app/src/main/java/mobile/halachipedia/MainActivity.java

```diff
@@ -1,29 +1,68 @@
 package mobile.halachipedia;
 
+import android.support.v4.widget.SwipeRefreshLayout;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.webkit.WebSettings;
-import android.webkit.WebView;
-import android.webkit.WebViewClient;
+import android.webkit.*;
 import android.view.Window;
 import android.view.WindowManager;
+import android.widget.Toast;
 
 public class MainActivity extends AppCompatActivity {
     private WebView webView;
+    private SwipeRefreshLayout swipe;
+    private final String URL = "https://www.halachipedia.com";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
         setContentView(R.layout.activity_main);
-        webView = (WebView) findViewById(R.id.webview);
-        webView.setWebViewClient(new WebViewClient());
-        if (savedInstanceState == null) {
-            webView.loadUrl("http://www.halachipedia.com");
-        }
+        loadWebView(savedInstanceState);
+        loadWebSettings();
+        loadSwipe();
+
+    }
+
+    private void loadWebSettings() {
         WebSettings webSettings = webView.getSettings();
         webSettings.setJavaScriptEnabled(true);
+        webSettings.setDomStorageEnabled(true);
+    }
+
+    private void loadWebView(Bundle savedInstanceState) {
+        webView = findViewById(R.id.webview);
+        webView.setWebViewClient(new WebViewClient(){
+            @Override
+            public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+                error();
+            }
+            @Override
```

Comment:
Are both of these overrides necessary? See https://stackoverflow.com/questions/6392318/detecting-webview-error-and-show-message/37145639#comment93292260_37145639